### DISCUSSION TITLE
Additional document customisation options

### DISCUSSION THREAD

**Discussion by aaronparker** [almenscorner/IntuneCD#45] [Ideas]
The generated markdown document has some fixed content at the beginining (title, tenant info, generated on etc.). The intro can already be customised and does work; however, I was thinking about whether those other items could be customised by supporting additional input via a JSON file or JSON input. For example:

```json
{
    "Title": "MEM Documentation \n\n",
    "Tenant": "{tenantname} \n\n",
    "Updated": "Documentation generated on {current_date} \n\n"
}
```

https://github.com/almenscorner/IntuneCD/blob/c75564b6478b08668145ce86190ee8b03caf241f/src/IntuneCD/run_documentation.py#L145

**Comment by almenscorner** [comment]
Good idea Aaron, I will try to squeeze it in to the upcoming version ;)

**Reply by aaronparker** [reply]
Thanks. If you're setting priorities I don't want to add any pressure. There are ways around this at the moment by modifying the generated markdown.

**Reply by almenscorner** [reply]
I have a working example where you provide a JSON string as an argument `-j "{\"title\": \"Tobias documentation\", \"intro\": \"Created from PyCharm\",\"tenant\": \"awesome tenant\",\"updated\": \"Generated on: \"}"`, if you like this approach I will add it to the upcoming version :)

**Reply by aaronparker** [reply]
Looks good to me

**Reply by almenscorner** [reply]
Added in 0e26c0e9ec0c81970c038f58ad7ef38dea3e5dcf and will be merged soon

**Reply by almenscorner** [reply]
1.1.2 is live and feature added 🎉  #50
